EmailAddressFinder/src/main/java/com/mycompany/emailaddressfinder/EmailAddressFinder.java

```diff
@@ -0,0 +1,53 @@
+package com.mycompany.emailaddressfinder;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+public class EmailAddressFinder {
+
+    private Set<String> emailAddresses = new HashSet();
+
+    public static void main(String[] args) throws Exception {
+        EmailAddressFinder finder = new EmailAddressFinder();
+        finder.findEmailsOnDomain("http://www.iana.org");
```

Comment:
Hmm, I believe one of the requirements was that the CLI tool takes the domain name as an argument.